src/main/java/com/synopsys/integration/detect/lifecycle/boot/decision/AutoScanTypeDecider.java

```diff
@@ -0,0 +1,165 @@
+                    scanTypeEvidenceMap.put(DetectTool.SIGNATURE_SCAN, pathsCollection.signaturePaths);
+                }
+                return scanTypeEvidenceMap;
+            }
+        }
+        return Collections.EMPTY_MAP;
+    }
+    
+    private final Set<String> avoidAbsolutley = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            ".gitattributes", 
+            ".gitignore", 
+            ".github", 
+            ".git",
+            "__MACOSX",
+            ".DS_Store")));
+    
+    /**
+     * This should be replaced with an intuitive binary reader that can zip and upload a stream of image formats
+     * since a malicious binary can be embedded in an image file. The original paths to the 
+     * individual image files should be stored in a text file for the BDBA worked to 
+     * reference in its report.
+     */
+    private final Set<String> ignoreReluctantly = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            ".png", 
+            ".gif", 
+            ".ico",
+            ".bmp",
+            ".jpeg",
+            ".jpg")));
+    
+    private final MediaTypeRegistry mediaTypeRegistry = MediaTypeRegistry.getDefaultRegistry();
+    
+    private boolean shouldAvoidDirectory(String name) {
+        return avoidAbsolutley.contains(name);
+    }
+    
+    private boolean isEligibleFile(String name) {
+        for(String extension : ignoreReluctantly) {
+            if (name.toLowerCase().endsWith(extension)) {
+                return false;
+            }
+        }
+        return true;
+    }
+    
+    private boolean isBinary(File file) throws IOException {
+        Set<MediaType> mediaTypes = new HashSet<>();
+        MediaType mediaType = MediaType.parse(new Tika().detect(new ByteArrayInputStream(FileUtils.readFileToByteArray(file))));
+        while(mediaType != null) {
+            mediaTypes.addAll(mediaTypeRegistry.getAliases(mediaType));
+            mediaTypes.add(mediaType);
+            mediaType = mediaTypeRegistry.getSupertype(mediaType);
+        }
+        return mediaTypes.stream().noneMatch(x -> x.getType().equals("text"));
+    }
+    
+    private PathsCollection search(Path pathToSearch) {
+        PathsCollection pathsCollection = new PathsCollection();
+        long t1 = System.currentTimeMillis();
+        try {
+            Files.walkFileTree(pathToSearch, new FileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    if (isEligibleFile(file.toFile().getName())) {
+                        if (isBinary(file.toFile())) {
+                            pathsCollection.binaryPaths.add(file.toAbsolutePath().toString());
+                        } else if (pathsCollection.detectorPaths.isEmpty()) {
+                            pathsCollection.detectorPaths.add(pathToSearch.toAbsolutePath().toString());
```

Comment:
I believe this paths won't be used anywhere in the DetectRun execution. It is just a toggle for running Detector Tool right?

Reply:
For now, yes.